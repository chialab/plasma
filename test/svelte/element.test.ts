import { afterEach, describe, expect, test } from 'vitest';
import TestElement from './components/TestElement.svelte';

let host: HTMLElement;

describe('Element', () => {
    afterEach(() => {
        host.remove();
    });

    test('mount component', async () => {
        host = document.createElement('div');
        host.setAttribute('id', 'host');
        document.body.appendChild(host);
        const instance = new TestElement({
            target: host,
            props: {
                stringProp: 'test',
                booleanProp: true,
                numericProp: 1,
                objectProp: { test: true },
            },
        });
        expect(instance).toBeTruthy();
        expect(host.innerHTML).toMatchSnapshot();
    });
});
