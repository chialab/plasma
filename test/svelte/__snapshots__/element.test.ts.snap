// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Element > mount component 1`] = `"<test-element :scope=\\"test-element\\" :defined=\\"\\" stringprop=\\"test\\" booleanprop=\\"\\" numericprop=\\"1\\"><span class=\\"text\\">test</span><span class=\\"boolean\\">true</span><span class=\\"number\\">1</span><span class=\\"object\\">{\\"test\\":true}</span></test-element>"`;
